packages/mgt-components/src/components/mgt-todo/mgt-todo.ts

```diff
@@ -375,49 +414,23 @@ export class MgtTodo extends MgtTasksBase {
    * @memberof MgtTodo
    */
   protected renderTask = (task: TodoTask) => {
-    const taskClasses = classMap({
-      'read-only': this.readOnly,
-      task: true
-    });
+    const isCompleted = task.status === 'completed';
 
-    return html`
-      <fluent-checkbox 
-        id=${task.id}
-        class=${taskClasses}
-        ?disabled=${this.readOnly}
-        @click="${() => this.handleTaskCheckClick(task)}"
-      >
-        ${this.renderTaskDetails(task)}
-      </fluent-checkbox>
-    `;
-  };
-
-  /**
-   * Render a completed task in the list.
-   *
-   * @protected
-   * @param {TodoTask} task
-   * @returns {TemplateResult}
-   * @memberof MgtTodo
-   */
-  protected renderCompletedTask = (task: TodoTask) => {
     const taskClasses = classMap({
-      complete: true,
+      complete: isCompleted,
       'read-only': this.readOnly,
       task: true
     });
 
-    const taskCheckContent = html`${getSvg(SvgIcon.CheckMark)}`;
+    const taskCheckContent = isCompleted ? html`${getSvg(SvgIcon.CheckMark)}` : nothing;
 
     return html`
       <fluent-checkbox 
         id=${task.id} 
         class=${taskClasses} 
-        checked 
-        ?disabled=${this.readOnly} 
-        @click="${() => this.handleTaskCheckClick(task)}"
+        ?checked=${isCompleted}
       >
-        <div slot="checked-indicator">
+        <div slot="checked-indicator" @click="${() => this.handleTaskCheckClick(task)}">
```

Comment:
The click handler here needs to stay on the fluent-checkbox element, that allows it to respond to clicks on the label and the indicator, this is an accessibility concern, plus I don't think that putting a click handler on the slot element actually works.